### ISSUE TITLE
`CustomSelfDestructOperation` does not check for static frame

### ISSUE THREAD

**Issue by lukelee-sl** [hashgraph/hedera-services#14037]
### Problem

The `CustomSelfDestructOperation` does not descend from or use Besu implementation of SelfDestructOperation.  
We are missing a check for static frames
```
        if (frame.isStatic()) {
            return new Operation.OperationResult(cost, ExceptionalHaltReason.ILLEGAL_STATE_CHANGE);
```

### Solution

Add the check above to ensure the user gets the appropriate status code returned.

**Comment by MrValioBg** [comment]
Seems like this check already exists.
Services [reference here](https://github.com/hashgraph/hedera-services/blob/2f5ca38b2ceb8f5743edc8455c1ba011f942ddd6/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomSelfDestructOperation.java#L91-L92)
Besu [reference here](https://github.com/hyperledger/besu/blob/d2b42d56d84e41e2be0b77c0ba4c842fba027455/evm/src/main/java/org/hyperledger/besu/evm/operation/SelfDestructOperation.java#L72-L76)

Am I missing something?

**Comment by lukelee-sl** [comment]
> Seems like this check already exists. @lukelee-sl Services [reference here](https://github.com/hashgraph/hedera-services/blob/2f5ca38b2ceb8f5743edc8455c1ba011f942ddd6/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomSelfDestructOperation.java#L91-L92) Besu [reference here](https://github.com/hyperledger/besu/blob/d2b42d56d84e41e2be0b77c0ba4c842fba027455/evm/src/main/java/org/hyperledger/besu/evm/operation/SelfDestructOperation.java#L72-L76)
> 
> Is this it, or am I missing something?

@MrValioBg, Good point.  For one thing we do not descend from Besu's `SelfDestructOperation`.  I think I was wondering about [this](https://github.com/hashgraph/hedera-services/blob/2f5ca38b2ceb8f5743edc8455c1ba011f942ddd6/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomSelfDestructOperation.java#L103) line and wondering if this check is necessary if the check is done above as you rightly point out.

**Comment by david-bakin-sl** [comment]
Refactor of `customselfdestruct` mentioned above moved to #14756
